### PR TITLE
chore: disable GHA for changesets PR

### PR DESCRIPTION
**Pull request by TomasSlama**
### Description

We don't want to run tests in "Version Packages" PR from changesets and ideally, we want to be able merge "Version Packages" immediately.

- replaced yarn cache functionality, the old one never could find cache folder
- skip GHA
- skip Jenkins steps in build
- add [skip ci] to PR title automatically

**test run** - https://github.com/toptal/picasso/runs/4235712965?check_suite_focus=true
**test changeset PR** - https://github.com/toptal/picasso/pull/2251

### Screenshots
image

### Review

- [ ] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/api-principles.md)
- [ ] Annotate all `props` in component with documentation
- [ ] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure that tests pass by running `yarn test`
- [ ] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and
1. You have no extra requests.
2. You have optional requests.
   1. Add `nit: ` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because
1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:
1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

### CODE REVIEW

File: ci/jobs/picasso-pr-specs/Jenkinsfile [hunk 1]
```diff
@@ -25,6 +25,11 @@ pipeline {
   stages {
     // Perform this only for PRs
     stage('Git checkout PR') {
+      when {
+        not {
+          branch 'disable-github-actions-for-changesets'
```
Comment (denieler) [inline ci/jobs/picasso-pr-specs/Jenkinsfile:30]: probably it's a temporary name of the branch, right?

Comment (TomasSlama) [reply ci/jobs/picasso-pr-specs/Jenkinsfile:30]: yep, only for test purposes

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -73,7 +88,7 @@ jobs:
                 ]
               })
             }
-      - name: Add peerDependencies warning to "Version Package" PR
+      - name: Edit "Version Pacakge" PR
```
Comment (dmaklygin1) [inline .github/workflows/release.yml:91]: typo in a word `Package` :)

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -30,13 +30,28 @@ jobs:
         with:
           node-version: 14.x
 
-      - name: Check yarn cache
-        uses: c-hive/gha-yarn-cache@v2
```
Comment (dmaklygin1) [inline .github/workflows/release.yml:34]: Could you please explain what's wrong with this library, I'm not aware of it, thanks :)

**Review by dmaklygin1 — APPROVED** [review]

**Review by denieler — APPROVED** [review]

**Comment by TomasSlama** [comment]
@toptal-bot run visual

**Comment by TomasSlama** [comment]
@toptal-bot run all

**Pull request merged**